lib/src/net/coap_exchange.dart

```diff
@@ -219,7 +227,7 @@ class CoapKeyToken {
 class CoapKeyUri {
   /// Construction
   CoapKeyUri(this._uri, this._endpoint) {
-    _hash = _uri.hashCode * 31 + (_endpoint == null ? 0 : _endpoint.hashCode);
+    _hash = _uri.hashCode * 31 + (_endpoint?.hashCode ?? 0);
```

Comment:
I guess here we could use something like this?

```suggestion
    _hash = Object.hash(_uri, _endpoint);
```